src/main/scripts/src/app/view/view-controls/share-dialog/share-dialog.component.ts

```diff
@@ -0,0 +1,126 @@
+/*
+ * -----------------------------------------------------------------------\
+ * Lumeer
+ *
+ * Copyright (C) since 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * -----------------------------------------------------------------------/
+ */
+
+import {Component, EventEmitter, OnInit, Output} from '@angular/core';
+import {NotificationsService} from 'angular2-notifications/dist';
+import {UserService} from '../../../core/rest/user.service';
+import {User} from '../../../core/dto/user';
+
+const BACKSPACE_KEY = 8;
+const ENTER_KEY = 13;
+
+@Component({
+  selector: 'share-dialog',
+  templateUrl: './share-dialog.component.html',
+  styleUrls: ['./share-dialog.component.scss']
+})
+export class ShareDialogComponent implements OnInit {
+
+  @Output()
+  public closeDialog: EventEmitter<void> = new EventEmitter();
+
+  public emails: string[] = [];
+  public text = '';
+
+  public users: User[] = [];
+  public suggestions: string[];
+
+  public constructor(private notificationService: NotificationsService,
+                     private userService: UserService) {
+  }
+
+  public ngOnInit() {
+    this.userService.getUsers().subscribe((users: User[]) => {
+      this.users = users;
+    });
+  }
+
+  public onKeyUp(event: KeyboardEvent) {
+    switch (event.keyCode) {
+      case ENTER_KEY:
+        return;
+      case BACKSPACE_KEY:
+      default:
+        this.suggest();
+    }
+  }
+
+  public onKeyDown(event: KeyboardEvent) {
+    switch (event.keyCode) {
+      case BACKSPACE_KEY:
+        this.removeItem();
+        return;
+      case ENTER_KEY:
+        this.addItemOrShare();
+        return;
+    }
+  }
+
+  private removeItem() {
+    if (this.text === '') {
+      event.preventDefault();
```

Comment:
Where is the `event` defined?

Reply:
It is a shortcut for `window.event` object.